sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -874,6 +877,74 @@ class RocksDBSuite extends AlsoTestWithChangelogCheckpointingEnabled with Shared
     )
   }
 
+  testWithChangelogCheckpointingEnabled("RocksDBFileManager: deepCopy") {
+    withTempDir { dir =>
+      val dfsRootDir = dir.getAbsolutePath
+      val originalFileManager = new RocksDBFileManager(
+        dfsRootDir, Utils.createTempDir(), new Configuration)
+      val copiedFileManager = originalFileManager.deepCopy()
+
+      // Save a version of checkpoint files
+      val cpFiles = Seq(
+        "001.sst" -> 10,
+        "002.sst" -> 10,
+        "003.sst" -> 10
+      )
+      saveCheckpointFiles(originalFileManager, cpFiles, 1, 101)
+
+      // Ensure checkpoint metrics are different
+      assert(originalFileManager.latestSaveCheckpointMetrics.filesCopied == 3L)
+      assert(originalFileManager.latestSaveCheckpointMetrics.bytesCopied == 30L)
+      assert(copiedFileManager.latestSaveCheckpointMetrics.filesCopied == 0L)
+      assert(copiedFileManager.latestSaveCheckpointMetrics.bytesCopied == 0L)
+
+      // Checkpoint the same files
+      saveCheckpointFiles(originalFileManager, cpFiles, 2, 101)
+      saveCheckpointFiles(copiedFileManager, cpFiles, 2, 101)
+
+      // Original file manager should skip since files already uploaded
+      assert(originalFileManager.latestSaveCheckpointMetrics.filesCopied == 0L)
+      assert(originalFileManager.latestSaveCheckpointMetrics.bytesCopied == 0L)
+
+      // Copied file manager should not skip since these are new files
+      assert(copiedFileManager.latestSaveCheckpointMetrics.filesCopied == 3L)
+      assert(copiedFileManager.latestSaveCheckpointMetrics.bytesCopied == 30L)
+    }
+  }
+
+  testWithChangelogCheckpointingEnabled("RocksDBFileManager: eliminate lock contention") {
```

Comment:
Can we be more specific and name it "background snapshot upload doesn't acquire rocksdb instance lock"?